shell/browser/api/electron_api_view.cc

```diff
@@ -280,10 +282,78 @@ void View::RemoveChildView(gin_helper::Handle<View> child) {
   }
 }
 
-void View::SetBounds(const gfx::Rect& bounds) {
+void View::SetBounds(const gfx::Rect& bounds, gin::Arguments* const args) {
+  bool animate = false;
+  int duration = 250;
+  std::string easing = "linear";
+
+  gin_helper::Dictionary options;
+  if (args->GetNext(&options)) {
+    options.Get("animate", &animate);
+    options.Get("duration", &duration);
+    options.Get("easing", &easing);
+  }
+
+  if (duration < 0)
+    duration = 0;
+
   if (!view_)
     return;
-  view_->SetBoundsRect(bounds);
+
+  if (!animate) {
+    view_->SetBoundsRect(bounds);
+
+    return;
+  }
+
+  gfx::Tween::Type tween_type = gfx::Tween::LINEAR;
+
+  if (easing == "ease-in") {
```

Comment:
this would be replaced above with something like:

```
template <>
struct Converter<gfx::Tween::Type> {
  static bool FromV8(v8::Isolate* isolate,
                     v8::Local<v8::Value> val,
                     gfx::Tween::Type* out) {
    std::string easing = base::ToLowerASCII(gin::V8ToString(isolate, val));
    if (orientation == "linear") {
      *out = gfx::Tween::LINEAR;
    } else if (orientation == "ease-in") {
      *out = gfx::Tween::EASE_IN;
    } else if (orientation == "ease-out") {
      *out = gfx::Tween::EASE_OUT;
    } else if (orientation == "ease-in-out") {
      *out = gfx::Tween::EASE_IN_OUT;
    } else {
      return false;
    }
    return true;
  }
};
```